### ISSUE TITLE
Job succesfully finishes in case if Pillar can't be compiled

### ISSUE THREAD

**Issue by vitalyisaev2** [jenkinsci/saltstack-plugin#28]
Hello!

I would suggest that the failure of SaltStack job caused by uncompilable Pillar should be unambiguously interpreted as an error on Jenkins.

Using ```1.6.1``` version of plugin I'm encountering this undesirable behaviour. 

```
Started by user jenkins
[EnvInject] - Loading node environment variables.
Building remotely on deployer (deploy) in workspace /var/lib/jenkins/workspace/DEPLOY_postgresql_avatars_migrate
 > git rev-parse --is-inside-work-tree # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url gitlab@gitlab.pio.tv.ro:Database/avatars.git # timeout=10
Fetching upstream changes from gitlab@gitlab.pio.tv.ro:Database/avatars.git
 > git --version # timeout=10
 > git -c core.askpass=true fetch --tags --progress gitlab@gitlab.pio.tv.ro:Database/avatars.git +refs/heads/*:refs/remotes/origin/*
 > git rev-parse refs/remotes/origin/master^{commit} # timeout=10
 > git rev-parse refs/remotes/origin/origin/master^{commit} # timeout=10
Checking out Revision cb7dcdff99f4620aaad431f8bc190dafe2b8755d (refs/remotes/origin/master)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f cb7dcdff99f4620aaad431f8bc190dafe2b8755d
 > git rev-list cb7dcdff99f4620aaad431f8bc190dafe2b8755d # timeout=10
Sending JSON: [{"client":"local","tgt":"p-3-testmap.pio.tv.ro","expr_form":"glob","fun":"state.sls","arg":["mailpaas.postgresql.migrate.avatars"]}]
Response on state.sls mailpaas.postgresql.migrate.avatars for p-3-testmap.pio.tv.ro:
[{"p-3-testmap.pio.tv.ro": [
  "Pillar failed to render with the following messages:",
  "Rendering SLS 'nginx-frontend' failed, render error:\nJinja error: oauth2.sls\nTraceback (most recent call last):\n  File \"/usr/lib/python2.7/dist-packages/salt/utils/templates.py\", line 366, in render_jinja_tmpl\n    output = template.render(**decoded_context)\n  File \"/usr/lib/python2.7/dist-packages/jinja2/environment.py\", line 989, in render\n    return self.environment.handle_exception(exc_info, True)\n  File \"/usr/lib/python2.7/dist-packages/jinja2/environment.py\", line 754, in handle_exception\n    reraise(exc_type, exc_value, tb)\n  File \"<template>\", line 18, in top-level template code\n  File \"/usr/lib/python2.7/dist-packages/salt/utils/jinja.py\", line 144, in get_source\n    raise TemplateNotFound(template)\nTemplateNotFound: oauth2.sls\n\n; line 18\n\n---\n[...]\n{% set av = av_cfg.avatars %}\n\n{% import_yaml \"attachloader.sls\" as al_cfg %}\n{% set al = al_cfg.attachloader %}\n\n{% import_yaml \"oauth2.sls\" as oauth2_cfg %}    <======================\n{% set oauth2 = oauth2_cfg.oauth2 %}\n\n{% import_yaml \"raw-content-reader.sls\" as rcr_cfg %}\n{% set rcr = rcr_cfg[\"raw-content-reader\"] %}\n\n[...]\n---"
]}]
Warning: you have no plugins providing access control for builds, so falling back to legacy behavior of permitting any downstream builds to be triggered
Triggering a new build of DEPLOY_cocaine_runtime
Finished: SUCCESS
```

Pleas consider this issue.

**Comment by mchugh19** [comment]
Are you able to replicate this issue on 1.7.1? This was likely fixed in 0213ec9 which was released in 1.7.0.

**Comment by vitalyisaev2** [comment]
@mchugh19 ```1.7.1``` version works correctly. Thank you!
